src/spec.c

```diff
@@ -1520,7 +1507,6 @@ void IndexSpec_InitializeSynonym(IndexSpec *sp) {
 
 IndexSpec *NewIndexSpec(const char *name) {
   IndexSpec *sp = rm_calloc(1, sizeof(IndexSpec));
-  sp->fields = rm_calloc(sizeof(FieldSpec), SPEC_MAX_FIELDS);
```

Comment:
TODO: Should a new empty schema (clone from null) should have `SPEC_MAX_FIELDS` fields allocated?